code/datums/flock/flock.dm

```diff
@@ -870,4 +898,12 @@ var/flock_signal_unleashed = FALSE
 		// get next turf
 		T = locate(ox + x, oy + y, z)
 
-
+// search for any radio device, starting with hands and then equipment
+// anything else is arbitrarily too deeply hidden and stowed away to get the signal
+// (more practically, they won't hear it)
+/proc/find_radio_on(mob/M)
```

Comment:
Since this is being moved to a general proc, might be better to rename it to clarify its only finding radios on mobs, also an early return on !istype(M) might be safe

Reply:
I mean it takes a mob argument, I think it's clear enough. Will add the type check tho.